client/gui/console.cpp

```diff
@@ -1,254 +1,226 @@
-	Console::Console() : buf{ new tec::RingBuffer< std::tuple< ImVec4, std::string >, 4096>() } {
-		inputBuf[0] = '\0';
-		
-		// Default embed commands
-		this->AddConsoleCommand( "cmdlist",
-			"cmdlist : List all commands",
-		[this ] (const char*) {
-			for (const auto& command : this->commands) {
-				this->Println(command.first);
-			}
-		});
-		
-		this->AddConsoleCommand( "help",
-			"help [command] : Prints a short help about an command",
-		[this ] (const char* args) {
-			const char* end_arg = args;
-			while (*end_arg != '\0' && *end_arg != ' ') {
-				end_arg++;
-			}
-			// Args now points were the arguments begins
-			std::string command(args, end_arg - args);
-			auto search = commands.find(command);
-			if(search != commands.end()) {
-				this->Println(std::get<1>(search->second));
-			} else {
-				this->Println("Unknown command. Please use cmdlist to list all commands.");
-			}
-		});
-		
-		this->AddConsoleCommand( "clear",
-			"clear : Clear console output",
-		[this ] (const char*) {
-			this->Clear();
-		});
-		
-		this->AddConsoleCommand( "echo",
-			"echo [message] : Prints a message to the console",
-		[this ] (const char* args) {
-			this->Println(args);
-		});
-		
-	}
+Console::Console() : buf{new tec::RingBuffer<std::tuple<ImVec4, std::string>, 4096>()} {
+	inputBuf[0] = '\0';
 
-	void Console::Update(double) {
-		EventQueue<WindowResizedEvent>::ProcessEventQueue();
-		EventQueue<KeyboardEvent>::ProcessEventQueue();
-	}
+	// Default embed commands
+	this->AddConsoleCommand("cmdlist", "cmdlist : List all commands", [this](const char*) {
+		for (auto command : this->commands) {
+			this->Println(command.first);
+		}
+	});
 
-	void Console::Clear() {
-		buf->clear();
-	}
+	this->AddConsoleCommand("help", "help [command] : Prints a short help about an command", [this](const char* args) {
+		const char* end_arg = args;
+		while (*end_arg != '\0' && *end_arg != ' ') {
+			end_arg++;
+		}
+		// Args now points were the arguments begins
+		std::string command(args, end_arg - args);
+		auto search = commands.find(command);
+		if (search != commands.end()) {
+			this->Println(std::get<1>(search->second));
+		}
+		else {
+			this->Println("Unknown command. Please use cmdlist to list all commands.");
+		}
+	});
 
-	void Console::Println(const std::string& str, ImVec4 color) {
-		{
-			std::lock_guard<std::mutex> lg(input_mutex);
-			if (buf->full()) {
-				buf->pop_back();
-			}
-			buf->push_front(std::make_tuple(color, str));
+	this->AddConsoleCommand("clear", "clear : Clear console output", [this](const char*) { this->Clear(); });
+
+	this->AddConsoleCommand("echo", "echo [message] : Prints a message to the console", [this](const char* args) {
+		this->Println(args);
+	});
+}
+
+void Console::Update(double) {
+	EventQueue<WindowResizedEvent>::ProcessEventQueue();
+	EventQueue<KeyboardEvent>::ProcessEventQueue();
+}
+
+void Console::Clear() { buf->clear(); }
+
+void Console::Println(const std::string& str, ImVec4 color) {
+	{
```

Comment:
This scope is great, but nearly useless as the only additional statement in the containing scope is an assignment.

Reply:
um, yeah, that seems excessive.